### DISCUSSION TITLE
Node Heterogeneous Hypergraph Implementation

### DISCUSSION THREAD

**Discussion by sheyril** [pyg-team/pytorch_geometric#5171] [Q&A]
Hi, I'm trying to carry out node classification using HypergraphConv provided by the library. I'm wondering if the implementation will work with heterogeneous nodes (I don't necessarily need to use attention at this point)? 

The [implemented paper](https://arxiv.org/pdf/1901.08150.pdf) only mentions heterogeneity in the context of attention. From what I see in the docs, the forward function expects me to input a node feature matrix of dimension: (Number of nodes, Number of Features), suggesting homogeneous nodes. 

I'm quite new to both Hypergraphs and Heterogeneous data, I'd appreciate any help. Thank you for your time!

**Comment by wsad1** [comment]
I haven't tested this.
But maybe you could use [HeteroConv](https://pytorch-geometric.readthedocs.io/en/latest/modules/nn.html#torch_geometric.nn.conv.HeteroConv). For each `edge_type` you could pass the `HypergraphConv`.

**Comment by rusty1s** [comment]
What you can do is store incidence matrices instead of adjacency matrices within `HeteroData`, e.g., a node type now corresponds to a set of hyperedges. You should then be able to leverage regular GNN layers to process.
